tracer/src/Datadog.Trace/ClrProfiler/CommonTracer.cs

```diff
@@ -0,0 +1,56 @@
+// <copyright file="CommonTracer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler
+{
+    /// <summary>
+    /// This class contains methods implemented by both the automatic and manual tracer.
+    /// It is used for duplex communication.
+    /// </summary>
+    internal abstract class CommonTracer : ICommonTracer
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(CommonTracer));
+
+        public void LockSamplingPriority()
+        {
+            var traceContext = Tracer.Instance.ActiveScope?.Span.Context?.TraceContext;
+
+            if (traceContext != null)
+            {
+                traceContext.LockSamplingPriority(notifyDistributedTracer: false);
+            }
+        }
+
+        public int? TrySetSamplingPriority(int? samplingPriority)
+        {
+            var traceContext = Tracer.Instance.ActiveScope?.Span.Context?.TraceContext;
+
+            // If there is no trace context, when a new span is propagated the sampling priority will automatically be locked
+            // because it will be considered as a distributed trace
```

Comment:
We may not need this some of this complexity (I am calling it that because I don't completely understand it). I learned recently that different parts of a distributed trace are allowed to make different sampling decisions. It's part of the work I'd doing where each part will report how the decision was done.

Reply:
I don't know if we can really apply that reasoning here, seeing as our use of "distributed" tracing is really a bit of a hack, it's not _actually_ a distributed trace. I'm in favour of anything that reduces complexity in general of course!